presto-main/src/test/java/com/facebook/presto/operator/scalar/TestIntervalFunctions.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.type.SqlIntervalDayTime;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.type.IntervalDayTimeType.INTERVAL_DAY_TIME;
+
+public class TestIntervalFunctions
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testIntervalFromDuration()
+    {
+        assertFunction("interval('1234 ns')", INTERVAL_DAY_TIME, new SqlIntervalDayTime(0, 0, 0, 0, 0));
+        assertFunction("interval('1234 ms')", INTERVAL_DAY_TIME, new SqlIntervalDayTime(0, 0, 0, 1, 234));
```

Comment:
To cover all the units supported by `Duration` and for the sake of completeness, we can also add tests for `us` (microseconds), although `SqlIntervalDayTime` currently cannot represent it.